homeassistant/components/xiaomi_miio/vacuum.py

```diff
@@ -399,6 +403,8 @@ async def async_clean_segment(self, segments):
         """Clean the specified segments(s)."""
         if isinstance(segments, int):
             segments = [segments]
+        elif isinstance(segments, str):
+            segments = list(map(int, segments.split(",")))
```

Comment:
I personally would consider this bad practice. Why not provide a list of integers instead?